org-make-toc.el

```diff
@@ -329,8 +329,11 @@ with the destination of the published file."
                                  (seq-intersection org-make-toc-exclude-tags (org-get-tags))
                                  ;; NOTE: The "COMMENT" keyword is not returned as the to-do keyword
                                  ;; by `org-heading-components', so it can't be tested as a keyword.
-                                 (when-let ((headline-text ((nth 4 (org-heading-components)))))
-                                   (string-match-p (rx bos "COMMENT" (or blank eos)) headline-text)))
+                                 (if-let ((headline-text (nth 4 (org-heading-components))))
+                                     (string-match-p (rx bos "COMMENT" (or blank eos)) headline-text)
+                                   (delay-warning 'org-make-toc
```

Comment:
I didn't understand the manual or docstring's explanation of the TYPE argument to `delay-warning`, but use-package macros expand such that they use the package name symbol when calling `display-warning`.  As far as I can tell, any symbol is accepted, and the output to the user looks informative when done this way.